src/hooks/useCreatePoolHandler.tsx

```diff
@@ -1 +1,88 @@
-export const useCreatePoolHandler = () => {}
+import { useContext, useEffect, useState } from 'react'
+import { ethers, Contract } from 'ethers'
+import { WalletContext } from '../../src/context/WalletContext'
+import GammaPoolFactory from '../../abis/v1-core/GammaPoolFactory.sol/GammaPoolFactory.json'
+import toast from 'react-hot-toast'
+import { calcPoolKey } from '../../src/utils/getSmartContract'
+
+export const useCreatePoolHandler = () => {
+
+  const [gammaPoolFactory, setGammaPoolFactory] = useState<Contract | null>(null)
+  const [token1Addr, setToken1Addr] = useState<string>('')
+  const [token2Addr, setToken2Addr] = useState<string>('')
+  const [cfmmAddr, setCfmmAddr] = useState<string>('')
+
+  const { accountInfo, provider, signer } = useContext(WalletContext)
+  useEffect(() => {
+    async function fetchContract() {
+      if (!provider) {
+        console.log('Please connect wallet.')
+        return
+      }
+
+      let gammaFactoryaddr = process.env.NEXT_PUBLIC_GAMMAFACTORY_ADDR || ''
+      let _gammaPoolFactory = new ethers.Contract(
+        gammaFactoryaddr,
+        GammaPoolFactory.abi,
+        accountInfo && accountInfo?.address ? signer : provider
+      )
+
+      if (_gammaPoolFactory) {
+        setGammaPoolFactory(_gammaPoolFactory)
+      }
+    }
+    fetchContract()
+  }, [provider])
+
+  const createPool = async () => {
+    try {
+      if (!accountInfo || !gammaPoolFactory) {
+        toast.error("Please connect your wallet.")
+        return
+      }
+
+      // check if pool already exists
+      let pool = await gammaPoolFactory.getPool(calcPoolKey(cfmmAddr))
+      if (pool != ethers.constants.AddressZero) {
+        toast.error("Pool already exsts at " + pool)
+        return
+      }
+
+      let CreatePoolParams = {
+        cfmm: cfmmAddr,
+        tokens: [token1Addr, token2Addr],
+        protocol: 1,
```

Comment:
yeah don't hardcode protocol number here. User should choose a protocol. It's just for now the only protocol she can choose is UniswapV2, which will be protocol 1

Reply:
so in the client side, the user would choose the protocol by selecting the type of AMM they want to create the pool for (i.e. Balancer, Curve, UniSwapV2, V3, etc.)? I'm visualizing this in the form of a dropdown.

Reply:
I moved the protocol out as a variable to be selected when we add new protocols.

Reply:
Also started a drop down so it looks like the user has a selection but the only option is Uniswap V2 now.